wear/src/main/java/io/homeassistant/companion/android/home/HomeActivity.kt

```diff
@@ -61,6 +61,28 @@ class HomeActivity : ComponentActivity(), HomeView {
         SensorWorker.start(this)
 
         mainViewModel.initAllSensors()
+
+        if (mainViewModel.loadingState.value == MainViewModel.LoadingState.READY) {
+            lifecycleScope.launch {
+                launch {
+                    mainViewModel.entityRegistry = presenter.getEntityRegistry()
```

Comment:
I don't understand why we need to update the view model variables directly from the activity. Couldn't all these `launch` blocks be added to a single function in the view model? The activity doesn't care about the registries, just that the data is up-to-date.

If you want to be able to cancel it (using `lifecycleScope` right now so this still doesn't happen automatically on pause) you could assign `lifecycleScope.launch { ... }` to a `val` and cancel it in `onPause`.

Reply:
Ah good point, yes they should be in the view model and the scope doesn't matter here as its a one time pull, will update to move the methods :)